From 0f9f71e20e9c75aa83081fec6346d0310b32109c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:06:20 -0400
Subject: [usb] uas: Use all available stream ids

Message-id: <1410887463-20674-309-git-send-email-dzickus@redhat.com>
Patchwork-id: 95420
O-Subject: [RHEL7 PATCH 308/591] uas: Use all available stream ids
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit d3f7c1560aee57d0ec293253e0c0e79a84ea3016
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Wed Oct 23 17:46:17 2013 +0100

    uas: Use all available stream ids

    If we get ie 16 streams we can use stream-id 1-16, not 1-15.

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Sarah Sharp <sarah.a.sharp@linux.intel.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/storage/uas.c b/drivers/usb/storage/uas.c
index 33f9dcd..3f021f2 100644
--- a/drivers/usb/storage/uas.c
+++ b/drivers/usb/storage/uas.c
@@ -722,7 +722,7 @@ static int uas_eh_task_mgmt(struct scsi_cmnd *cmnd,
 {
  struct Scsi_Host *shost = cmnd->device->host;
  struct uas_dev_info *devinfo = (void *)shost->hostdata[0];
- u16 tag = devinfo->qdepth - 1;
+ u16 tag = devinfo->qdepth;
  unsigned long flags;
 
  spin_lock_irqsave(&devinfo->lock, flags);
@@ -843,7 +843,7 @@ static int uas_slave_configure(struct scsi_device *sdev)
 {
  struct uas_dev_info *devinfo = sdev->hostdata;
  scsi_set_tag_type(sdev, MSG_ORDERED_TAG);
- scsi_activate_tcq(sdev, devinfo->qdepth - 3);
+ scsi_activate_tcq(sdev, devinfo->qdepth - 2);
  return 0;
 }
 
@@ -1027,7 +1027,7 @@ static int uas_probe(struct usb_interface *intf, const struct usb_device_id *id)
  INIT_LIST_HEAD(&devinfo->dead_list);
  uas_configure_endpoints(devinfo);
 
- result = scsi_init_shared_tag_map(shost, devinfo->qdepth - 3);
+ result = scsi_init_shared_tag_map(shost, devinfo->qdepth - 2);
  if (result)
   goto free_streams;
 
-- 
1.7.1